### PR TITLE
unsupported-blocks-plugin 

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/plugin-commons/web/src/Base/createBasePlugin.ts [hunk 1]
```diff
@@ -62,6 +65,7 @@ interface CreateBasePluginConfig extends CreatePluginConfig {
   onOverlayClick?: ({ e, pubsub }: { e: Event; pubsub: Pubsub }) => void;
   onComponentMount?: ({ e, pubsub }: { e: Event; pubsub: Pubsub }) => void;
   disableRightClick?: UISettings['disableRightClick'];
+  supportedPluginsOnOneApp?: string[];
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBasePlugin.ts:68]: supportedBlockTypes

File: packages/plugin-commons/web/src/Base/createBasePlugin.ts [hunk 2]
```diff
@@ -250,9 +254,26 @@ const createBasePlugin = (
       // TODO subject to change for draft-js next release
       const contentState = getEditorState().getCurrentContent();
       const key = contentBlock.getEntityAt(0);
+
       if (key) {
         const entity = contentState.getEntity(key);
         const type = entity.getType();
+        if (
+          type === UNAVAILABLE_ON_ONEAPP_TYPE ||
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBasePlugin.ts:262]: delete line

File: packages/plugin-commons/web/package.json [hunk 3]
```diff
@@ -37,7 +37,8 @@
     "react-colorful": "^4.4.3",
     "react-measure": "~2.3.0",
     "wix-rich-content-common": "8.10.1",
-    "wix-rich-content-editor-common": "8.10.1"
+    "wix-rich-content-editor-common": "8.10.1",
+    "wix-rich-content-plugin-unavailable-on-oneapp": "8.10.1"
```
Comment (shaulgo) [inline packages/plugin-commons/web/package.json:41]: delete line

File: packages/plugin-commons/web/src/Base/createBasePlugin.ts [hunk 4]
```diff
@@ -250,9 +254,26 @@ const createBasePlugin = (
...
+            props: {
+              getData: getData(contentBlock, { getEditorState }),
+              setData: setData(contentBlock, { getEditorState, setEditorState }),
+              deleteBlock: deleteEntity(contentBlock, { getEditorState, setEditorState }),
+              type: UNAVAILABLE_ON_ONEAPP_TYPE,
+              unsupportedType: type,
+            },
+          };
+        }
         if (config.type === type || config.legacyType === type) {
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBasePlugin.ts:277]: let type
if (config.type === entityType || config.legacyType === entityType) {
   type = entityType;
else if (!config.supportedPluginsOnOneApp?.includes(type)) {
   type = UNAVAILABLE_ON_ONEAPP_TYPE;
}
const blockRenderObject = type ? {
            component: DecoratedCompWithBase,
            editable: false,
            props: {
              getData: getData(contentBlock, { getEditorState }),
              setData: setData(contentBlock, { getEditorState, setEditorState }),
              deleteBlock: deleteEntity(contentBlock, { getEditorState, setEditorState }),
              type,
            },
          } : null;
return blockRenderObject;

File: packages/plugin-unavailable-on-oneapp/web/src/defaults.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+// import { PaletteColors, ThemeUtils, RicosCustomStyles } from 'wix-rich-content-common';
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/defaults.ts:1]: clean up

File: packages/plugin-unavailable-on-oneapp/web/src/editor.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/*
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/editor.ts:1]: cleanup

File: packages/plugin-unavailable-on-oneapp/web/src/icons/CircleInfoIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+              </g>
+            </g>
+          </g>
+        </g>
+      </g>
+    </g>
+  </svg>
+
+  //   <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" {...props}>
+  //     <defs>
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/icons/CircleInfoIcon.tsx:41]: cleanup

File: packages/plugin-unavailable-on-oneapp/web/src/modals.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+// This is your plugin's ModalSetting & ModalMap.
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/modals.ts:1]: remove file

File: packages/plugin-unavailable-on-oneapp/web/src/toolbar/createToolbar.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import createInlineButtons from './inline-buttons';
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/toolbar/createToolbar.ts:1]: check if this is needed

File: packages/plugin-unavailable-on-oneapp/web/src/toolbar/inline-buttons.tsx [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { BUTTONS } from 'wix-rich-content-plugin-commons';
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/toolbar/inline-buttons.tsx:1]: check if needed

File: packages/plugin-unavailable-on-oneapp/web/src/toolbar/insert-buttons.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
+// import { DEFAULTS } from '../unavailable-on-oneapp-component';
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/toolbar/insert-buttons.tsx:1]: remove file

File: packages/plugin-unavailable-on-oneapp/web/src/typeMapper.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import UnavailableOnOneAppViewer from './unavailable-on-oneapp-viewer';
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/typeMapper.ts:1]: this file is probably not needed

File: packages/plugin-unavailable-on-oneapp/web/src/types.ts [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import { EditorPluginConfig, ViewerPluginConfig } from 'wix-rich-content-common';
+
+export const UNAVAILABLE_ON_ONEAPP_TYPE = 'unavailableononeapp';
+
+export interface UnavailableOnOneAppPluginEditorConfig extends EditorPluginConfig {
+  [key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+export interface UnavailableOnOneAppPluginViewerConfig extends ViewerPluginConfig {
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/types.ts:9]: probably don't need

File: packages/plugin-unavailable-on-oneapp/web/src/unavailable-on-oneapp-viewer.tsx [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    return unSupportedNames.slice(2, unSupportedNames.length).join('-');
+  };
+
+  render() {
+    this.styles = this.styles || mergeStyles({ styles, theme: this.props.theme });
+    const pluginName = this.getPluginName(this.props.unsupportedType);
+
+    return (
+      <div className={styles.unaOnOneApp_container}>
+        <div className={styles.unaOnOneApp_alert}>
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/unavailable-on-oneapp-viewer.tsx:28]: no need for 2 divs

File: packages/plugin-unavailable-on-oneapp/web/src/unavailable-on-oneapp-viewer.tsx [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+  styles: Record<string, string>;
+  getPluginName = fullPluginName => {
+    if (!fullPluginName.includes('-')) return fullPluginName;
+    const unSupportedNames = fullPluginName.split('-');
+    return unSupportedNames.slice(2, unSupportedNames.length).join('-');
+  };
+
+  render() {
+    this.styles = this.styles || mergeStyles({ styles, theme: this.props.theme });
+    const pluginName = this.getPluginName(this.props.unsupportedType);
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/unavailable-on-oneapp-viewer.tsx:24]:
```suggestion
    const pluginName = t(this.props.unsupportedType) || t('unsupportedBlockFallback');
```

File: packages/plugin-unavailable-on-oneapp/web/src/viewer.ts [hunk 16]
```diff
@@ -0,0 +1,22 @@
+/*
+  This module exports the required CreatePluginFunction for RicosViewer.
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/src/viewer.ts:2]: file not needed

File: packages/plugin-unavailable-on-oneapp/web/viewer/package.json [hunk 17]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (shaulgo) [inline packages/plugin-unavailable-on-oneapp/web/viewer/package.json:1]: not needed

**Review by shaulgo — CHANGES_REQUESTED** [review]

**Comment by tamrat-bay** [comment]
This branch has problems.
The changes were merged in different branch.

**Pull request closed without merging**